src/packed_text.cpp

```
@@ -19,6 +19,8 @@ PackedText::PackedText(const std::string& pac_path, int64_t n_bases)
     if (fstat(fd_, &st) != 0)
         err_fatal(__func__, "failed to fstat '%s': %s", pac_path.c_str(), strerror(errno));
     map_bytes_ = (size_t)st.st_size;
+    if (n_bases < 0)
+        err_fatal(__func__, "negative n_bases (%lld) for pac file '%s'", (long long)n_bases, pac_path.c_str());
```

Cppcheck (2.21.0)

[style] The function 'length' is never used.

(unusedFunction)